### PR TITLE
adds top level `data` key

### PR DESCRIPTION
**Pull request by MarchandMD**
# What does this PR do? 
- adds top level `data` key to the response when making a successful request to `GET /tasks`

# How does this PR accomplish that? 
- see the git diff notes

### CODE REVIEW

File: dist/models/Groups.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ const GroupSchema = mongoose.Schema({
     required: true
   },
   description: {
-    type: String
+    type: String,
+    required: false
   }
```
Comment (MarchandMD) [inline dist/models/Groups.js:12]: adds `required: false` as a fail-safe

File: dist/routes/groups.js [hunk 2]
```diff
@@ -44,4 +44,5 @@ groupRouter.get("/", (request, response) => {
  * */
 groupRouter.get("/:id", (request, response) => {
   Groups.findById(request.params.id).then(resp => response.status(200).json(resp)).catch(err => response.status(400).json("Request Failed"));
-});
\ No newline at end of file
+});
+module.exports = groupRouter;
```
Comment (MarchandMD) [inline dist/routes/groups.js:48]: exports the `groupRouter` to expose it to `app.js`

File: dist/routes/tasks.js [hunk 3]
```diff
@@ -21,7 +21,9 @@ const Tasks = require("../models/Tasks");
  *        description: Request Failed
  * */
 router.get("/", (request, response) => {
-  Tasks.find().then(resp => response.status(200).json(resp)).catch(err => response.status(400).json("Request Failed"));
+  Tasks.find().then(resp => response.status(200).json({
+    data: resp
+  })).catch(err => response.status(400).json("Request Failed"));
```
Comment (MarchandMD) [inline dist/routes/tasks.js:26]: adds `data:` key at the top level to adhere to JSONAPI specs

File: src/routes/tasks.js [hunk 4]
```diff
@@ -20,7 +20,7 @@ const Tasks = require("../models/Tasks")
  * */
 router.get("/", (request, response) => {
     Tasks.find()
-        .then((resp) => response.status(200).json(resp))
+        .then((resp) => response.status(200).json({ data: resp }))
         .catch((err) => response.status(400).json("Request Failed"))
```
Comment (MarchandMD) [inline src/routes/tasks.js:24]: this is the actual file that was edited; then `npm run build` made all the other changes

**Comment by MarchandMD** [comment]
I requested reviews from both of you since this changes the structure of the JSON that comes back from an API call to `GET /tasks`

I may make more changes to different API call response data to adhere to JSONAPI specifications

I started learning about Jest and quickly discovered that I'm doing things wrong lol

**Review by msakr21 — APPROVED** [review]

**Pull request merged**